gateway/modules/eventing/operations.go

```diff
@@ -23,16 +26,53 @@ func (m *Module) IsEnabled() bool {
 }
 
 // QueueEvent queues a new event
-func (m *Module) QueueEvent(ctx context.Context, project, token string, req *model.QueueEventRequest) error {
+func (m *Module) QueueEvent(ctx context.Context, project, token string, req *model.QueueEventRequest) (interface{}, error) {
 	m.lock.RLock()
 	defer m.lock.RUnlock()
 
 	err := m.validate(ctx, project, token, req)
 	if err != nil {
-		return err
+		logrus.Errorf("error queueing event in eventing unable to validate - %s", err.Error())
+		return nil, err
+	}
+
+	batchID, err := m.batchRequests(ctx, []*model.QueueEventRequest{req})
+	if err != nil {
+		logrus.Errorf("error queueing event in eventing unable to batch requests - %s", err.Error())
+		return nil, err
+	}
+
+	// if true then wait for event response
+	if req.IsSynchronous {
+		responseChan := make(chan interface{})
+		defer close(responseChan) // close channel
+		defer m.eventChanMap.Delete(batchID)
+		m.eventChanMap.Store(batchID, eventResponse{time: time.Now(), response: responseChan})
+		for {
+			select {
+			case <-ctx.Done():
+				// clear channel
+				return nil, ctx.Err()
+			case result := <-responseChan:
+				return result, nil
+			}
+		}
+	}
```

Comment:
Skip this in two steps. Store the channel in the map before you call batch request. Wait on the channel after batchRequests returns successfully. You will have to check the `req.IsSynchronous` in either case